src/NServiceBus.Testing/TypeExtensions.cs

```diff
@@ -1,39 +1,48 @@
          {
-             var interfaceType = interfaceGenericType.MakeGenericType(messageType);
- 
-             if (!interfaceType.IsAssignableFrom(targetType))
+             var interfaceTypes = targetType.GetInterfaces()
+                 .Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == interfaceGenericType)
+                 .Where(i => i.GenericTypeArguments.First().IsAssignableFrom(messageType));
+
+             foreach (var interfaceType in interfaceTypes)
              {
-                 return null;
-             }
- 
-             var methodInfo = targetType.GetInterfaceMap(interfaceType).TargetMethods.FirstOrDefault();
-             if (methodInfo == null)
-             {
-                 return null;
-             }
- 
-             var target = Expression.Parameter(typeof(object));
-             var messageParam = Expression.Parameter(typeof(object));
-             var contextParam = Expression.Parameter(typeof(IMessageHandlerContext));
- 
-             var castTarget = Expression.Convert(target, targetType);
- 
-             var methodParameters = methodInfo.GetParameters();
-             var messageCastParam = Expression.Convert(messageParam, methodParameters.ElementAt(0).ParameterType);
- 
-             Expression body = Expression.Call(castTarget, methodInfo, messageCastParam, contextParam);
- 
-             return Expression.Lambda<Func<object, object, IMessageHandlerContext, Task>>(body, target, messageParam, contextParam).Compile();
+                var methodInfo = targetType.GetInterfaceMap(interfaceType).TargetMethods.FirstOrDefault();
+                if (methodInfo == null)
+                {
+                    yield break;
+                }
+
+                var target = Expression.Parameter(typeof(object));
+                var messageParam = Expression.Parameter(typeof(object));
+                var contextParam = Expression.Parameter(typeof(IMessageHandlerContext));
+
+                var castTarget = Expression.Convert(target, targetType);
+
+                var methodParameters = methodInfo.GetParameters();
+                var messageCastParam = Expression.Convert(messageParam, methodParameters.ElementAt(0).ParameterType);
+
+                Expression body = Expression.Call(castTarget, methodInfo, messageCastParam, contextParam);
+
+                yield return Expression.Lambda<Func<object, object, IMessageHandlerContext, Task>>(body, target, messageParam, contextParam).Compile();
+            }
          }
+
+         public static async Task InvokeSerially(this IEnumerable<Func<object, object, IMessageHandlerContext, Task>> invokers, object instance, object message, IMessageHandlerContext context)
+         {
+            foreach (var invocation in invokers)
+            {
+                await invocation(instance, message, context);
```

Comment:
I think it is actually needed here. Test Fxs can have synchronization contexts as well.